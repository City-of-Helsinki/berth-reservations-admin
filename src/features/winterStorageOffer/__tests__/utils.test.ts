import {
  getAllSectionsIdentifiers,
  getApplicationBoat,
  getWinterStorageArea,
  getWinterStoragePlaceData,
  isPlaceSuitableForBoat,
} from '../utils';
import {
  getMockBoat,
  getMockPlace,
  mockApplication,
  mockAreaData,
  mockBoatTypes,
  mockSectionsData,
} from '../__fixtures__/mockData';

describe('winterStorageOffer utils', () => {
  describe('getWinterStoragePlaceData', () => {
    it('maps data correctly', () => {
      expect(getWinterStoragePlaceData(mockAreaData)).toMatchSnapshot();
    });
  });

  describe('getAllSectionsIdentifiers', () => {
    it('maps data correctly', () => {
      expect(getAllSectionsIdentifiers(mockSectionsData)).toMatchSnapshot();
    });
  });

  describe('getWinterStorageArea', () => {
    it('maps data correctly', () => {
      expect(getWinterStorageArea(mockAreaData)).toMatchSnapshot();
    });
  });

  describe('getApplicationBoat', () => {
    it('maps data correctly', () => {
      expect(getApplicationBoat(mockApplication, mockBoatTypes));
    });
  });

  describe('isPlaceSuitableForBoat', () => {
    it('returns true if the boat is shorter or equal than the place', () => {
      expect(isPlaceSuitableForBoat(getMockPlace(3), getMockBoat(2))).toEqual(true);
      expect(isPlaceSuitableForBoat(getMockPlace(2), getMockBoat(2))).toEqual(true);
    });

    it('returns false if the boat is longer than the place', () => {
      expect(isPlaceSuitableForBoat(getMockPlace(2), getMockBoat(3))).toEqual(false);
    });

    it('returns true if only the boat length is unknown', () => {
      expect(isPlaceSuitableForBoat(getMockPlace(2), getMockBoat(null))).toEqual(true);
      expect(isPlaceSuitableForBoat(getMockPlace(2), undefined)).toEqual(true);
    });

    it('returns false if only the place length is unknown', () => {
      expect(isPlaceSuitableForBoat(getMockPlace(null), getMockBoat(2))).toEqual(false);
    });

    it('returns true if both lengths are unknown', () => {
      expect(isPlaceSuitableForBoat(getMockPlace(null), getMockBoat(null))).toEqual(true);
      expect(isPlaceSuitableForBoat(getMockPlace(null), undefined)).toEqual(true);
    });
  });
});
