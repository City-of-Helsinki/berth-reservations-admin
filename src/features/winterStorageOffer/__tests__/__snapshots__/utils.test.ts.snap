// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`winterStorageOffer utils getAllSectionsIdentifiers maps data correctly 1`] = `
Array [
  Object {
    "disabled": false,
    "label": "1",
    "value": "1",
  },
  Object {
    "disabled": true,
    "label": "2",
    "value": "2",
  },
]
`;

exports[`winterStorageOffer utils getWinterStorageArea maps data correctly 1`] = `
Object {
  "imageFile": "",
  "municipality": "Helsinki",
  "name": "AREA",
  "properties": Object {
    "electricity": true,
    "gate": true,
    "numberOfCustomers": 1,
    "summerStorageForBoats": true,
    "summerStorageForDockingEquipment": true,
    "summerStorageForTrailers": true,
    "water": true,
  },
  "streetAddress": "Testiosoite 1",
  "wwwUrl": "",
  "zipCode": "00100",
}
`;

exports[`winterStorageOffer utils getWinterStoragePlaceData maps data correctly 1`] = `
Array [
  Object {
    "area": "AREA",
    "areaId": "MOCK-AREA",
    "id": "MOCK-PLACE",
    "isActive": true,
    "isAvailable": false,
    "leases": Array [
      Object {
        "customer": Object {
          "id": "MOCK-PROFILE",
        },
        "endDate": undefined,
        "isActive": true,
        "startDate": undefined,
        "status": "PAID",
      },
    ],
    "length": 3,
    "number": 1,
    "properties": Object {
      "electricity": true,
      "gate": true,
      "summerStorageForBoats": true,
      "summerStorageForDockingEquipment": true,
      "summerStorageForTrailers": true,
      "water": true,
    },
    "section": "1",
    "width": 3,
  },
]
`;
